### ISSUE TITLE
Timestamp and Stored losing accuracy

### ISSUE THREAD

**Issue by ht2** [LearningLocker/learninglocker#772]
The use of `strtotime()` when casting string dates into mongo dates is losing the micro seconds accuracy that is present on ISO8601 timestamps (leading to conformance test failure when querying since/until values at the microsecond level).

Places where we are casting to MongoDate should employ the following technique to ensure that microsecond accuracy is kept when coming from a string. Using the Carbon library will ensure proper DateTime creation from any string format. Passing the `$carbonDate->micro` attribute on MongoDate creation will ensure the accuracy is kept.

Note that this patch would be best applied into the `issue/storedDate` branch where improvements to the date function have already been made.

```
$carbon1 = new Carbon\Carbon('2015-10-07T03:00');
$carbon2 = new Carbon\Carbon('2015-10-07T14:28:41.219Z');
$carbon3 = new Carbon\Carbon('2016-01-13T17:39:49.219000+00:00');

$mongodate1 = new \MongoDate($carbon1->timestamp, $carbon1->micro);
$mongodate2 = new \MongoDate($carbon2->timestamp, $carbon2->micro);
$mongodate3 = new \MongoDate($carbon3->timestamp, $carbon3->micro);
```

**Comment by ht2** [comment]
@andrewhickey has made good progress in https://github.com/LearningLocker/learninglocker/tree/issue/mongodates

I'llmake a PR to bring that in, and then we can look as to other outstanding areas

**Comment by ht2** [comment]
`issue/mongodates` PR'd into `issue/storedDate` via #774

**Comment by ht2** [comment]
https://github.com/LearningLocker/learninglocker/blob/issue/storedDate/app/locker/repository/Statement/EloquentIndexer.php#L48

and

https://github.com/LearningLocker/learninglocker/blob/issue/storedDate/app/locker/repository/Statement/EloquentIndexer.php#L58

..need amends

**Comment by ht2** [comment]
Fixed in v1.12.0
